code/game/atoms_movable.dm

```diff
@@ -771,3 +771,38 @@
 	if(anchored || throwing)
 		return FALSE
 	return TRUE
+
+
+/obj/item/proc/do_pickup_animation(turf/target)
+	set waitfor = FALSE
+	var/image/I = image(icon = src, loc = loc, layer = layer + 0.1)
+	I.plane = GAME_PLANE
+	I.transform *= 0.75
+	I.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA
+	var/direction = get_dir(src, target)
+	var/to_x = 0
+	var/to_y = 0
+	if(direction & NORTH)
+		to_y = 32
+	else if(direction & SOUTH)
+		to_y = -32
+
+	if(direction & EAST)
+		to_x = 32
+	else if(direction & WEST)
+		to_x = -32
+
+	if(!direction)
+		to_y = 16
+
+	flick_overlay(I, GLOB.clients, 6)
+	var/matrix/M = new
+	M.Turn(pick(-30, 30))
+
+	animate(I, transform = M, time = 1)
+	sleep(1)

```

Comment:
:clap: sanity :clap: checks :clap: for :clap: item :clap: still :clap: existing :clap: these :clap: are :clap: sleeps :clap:

Reply:
also if you're setting wait_for = false these sleeps are literally doing nothing anyway?

Reply:
think the waitfor is so it doesn't delay/impact the actual pickup

Reply:
the sleep is for animationn work